services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/registry.py

```diff
@@ -46,47 +50,94 @@ async def _is_registry_reachable(registry_settings: RegistrySettings) -> None:
             raise _RegistryNotReachableError(error_message)
 
 
-async def wait_for_registry_liveness(app: FastAPI) -> None:
-    registry_settings: RegistrySettings = app.state.settings.REGISTRY_SETTINGS
+async def _is_dockerhub_reachable(registry_settings: RegistrySettings) -> None:
+    async with httpx.AsyncClient(timeout=5) as client:
+        params: dict[str, Any] = {"headers": {"Content-Type": "application/json"}}
+        if registry_settings.REGISTRY_AUTH:
+            params["auth"] = (
+                registry_settings.REGISTRY_USER,
+                registry_settings.REGISTRY_PW.get_secret_value(),
+            )
+            params["data"] = json.dumps(
+                {
+                    "username": registry_settings.REGISTRY_USER,
+                    "password": registry_settings.REGISTRY_PW.get_secret_value(),
+                }
+            )
+
+        # NOTE: uses different URL than the one specified in the configuration
+        url = "https://hub.docker.com/v2/users/login"
+
+        _logger.info("Registry test url ='%s'", url)
+        response = await client.post(url, **params)
+        reachable = (
+            response.status_code == status.HTTP_200_OK and "token" in response.json()
+        )
+        if not reachable:
+            _logger.error("Response: %s %s", response, response.text)
+            error_message = (
+                f"Could not reach registry {response.request.url}"
+                f"auth={registry_settings.REGISTRY_AUTH}, because: {response.text}"
+            )
+            raise _RegistryNotReachableError(error_message)
+
+
+async def wait_for_registries_liveness(app: FastAPI) -> None:
+    settings: ApplicationSettings = app.state.settings
 
     await wait_for_service_liveness(
         _is_registry_reachable,
-        service_name="Registry",
-        endpoint=_get_registry_url(registry_settings),
-        registry_settings=registry_settings,
+        service_name="Internal Registry",
+        endpoint=_get_registry_url(settings.DY_DEPLOYMENT_REGISTRY_SETTINGS),
+        registry_settings=settings.DY_DEPLOYMENT_REGISTRY_SETTINGS,
     )
 
+    if settings.DY_DOCKER_HUB_REGISTRY_SETTINGS:
+        await wait_for_service_liveness(
+            _is_dockerhub_reachable,
+            service_name="DockerHub Registry",
+            endpoint="https://hub.docker.com/v2/users/login",
```

Comment:
See, here you repeat again the same endpoint above `https://hub.docker.com/v2/users/login`. I would highly recommend to add it in the settings